File: fapi/src/main/java/com/netcracker/curs/fapi/service/ComplaintService.java
package com.netcracker.curs.fapi.service;

import com.netcracker.curs.fapi.models.Complaint;

import javax.servlet.http.HttpServletRequest;
import java.util.List;

public interface ComplaintService {
    Complaint getComplaintById(Integer id, HttpServletRequest request);
    Complaint saveComplaint(Complaint post, HttpServletRequest request);
    void deleteComplaint(Integer id, HttpServletRequest request);
    void deletePost(Integer id, HttpServletRequest request);
    List<Complaint> getAll(HttpServletRequest request);

}

File: frontend/src/app/services/complaint.service.ts
import {Injectable} from "@angular/core";
import {HttpClient, HttpParams, HttpRequest} from "@angular/common/http";
import {Observable} from "rxjs";
import {Complaint} from "../models/Complaint";
import {RequestOptions, Headers} from "@angular/http";

@Injectable()
export class ComplaintService {

  constructor(private http: HttpClient) {
  }

  getComplaint(): Observable<Complaint[]> {
    return this.http.get<Complaint[]>('/api/complaint');
  }

  saveComplaint(complaint: Complaint): Observable<Complaint> {
    return this.http.post<Complaint>('/api/complaint', complaint);
  }

  deleteComplaint(complaintId: number): Observable<void> {
    return this.http.delete<void>( '/api/complaint/' + complaintId);
  }

  deletePost(postId: number): Observable<void> {
    return this.http.delete<void>('/api/post/' + postId);
  }

  getComplaintById(id: string): Observable<Complaint> {
    return this.http.get<Complaint>('/api/complaint/' + id);
  }
}
